src/replication.c

```diff
@@ -4352,7 +4354,19 @@ initialize_replication()
 void
 destroy_volume(spec_t *spec)
 {
-	ASSERT0(get_num_entries_from_mempool(&spec->rcommon_deadlist));
+    int ret = 0;
+    void *res;
+
+    destroy_rcommon_deadlist(spec);
```

Comment:
shouldn't we clean rcommon_deadlist after cancelling the thread. It's possible that thread is working on a pointer which is just freed by this destroy?

Reply:
thats a good one @pawanpraka1